packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,286 @@
+    pointing: PropTypes.bool,
+    position: PropTypes.oneOf(POSITIONS),
+    target: PropTypes.any,
+    trigger: customPropTypes.every([customPropTypes.disallow(['children']), PropTypes.element]),
+    content: customPropTypes.shorthandAllowingChildren,
+  }
+
+  static defaultProps: TooltipProps = {
+    align: 'center',
+    mountNode: isBrowser() ? document.body : null,
+    position: 'above',
+    mouseLeaveDelay: 500,
+    pointing: true,
+  }
+
+  static autoControlledProps = ['open']
+
+  pointerTargetRef = React.createRef<HTMLElement>()
+  triggerRef = React.createRef<HTMLElement>() as React.MutableRefObject<HTMLElement>
+  tooltipDomElement = null
+
+  closeTimeoutId
+
+  renderComponent({
+    classes,
+    rtl,
+    accessibility,
+  }: RenderResultConfig<TooltipProps>): React.ReactNode {
+    const { mountNode, children, trigger } = this.props
+    const tooltipContent = this.renderTooltipContent(classes.content, rtl, accessibility)
+
+    const triggerElement = React.Children.only(
+      childrenExist(children) ? children : trigger,
+    ) as React.ReactElement<any>
+    const triggerProps = this.getTriggerProps(triggerElement)
+
+    return (
+      <>
+        {triggerElement && (
+          <Ref innerRef={this.triggerRef}>
+            {React.cloneElement(triggerElement, {
+              ...accessibility.attributes.trigger,
+              ...triggerProps,
+              ...applyAccessibilityKeyHandlers(accessibility.keyHandlers.trigger, triggerProps),
+            })}
+          </Ref>
+        )}
+        {mountNode && ReactDOM.createPortal(tooltipContent, mountNode)}
+      </>
+    )
+  }
+
+  getTriggerProps(triggerElement) {
+    const triggerProps: Props = {}
+
+    triggerProps.onFocus = (e, ...args) => {
+      if (isFromKeyboard()) {
+        this.trySetOpen(true, e)
+      }
+      _.invoke(triggerElement, 'props.onFocus', e, ...args)
+    }
+    triggerProps.onBlur = (e, ...args) => {
+      if (this.shouldClose(e)) {
+        this.trySetOpen(false, e)
+      }
+      _.invoke(triggerElement, 'props.onBlur', e, ...args)
+    }
+
+    triggerProps.onMouseEnter = (e, ...args) => {
+      this.setTooltipOpen(true, e)
+      setWhatInputSource('mouse')
+      _.invoke(triggerElement, 'props.onMouseEnter', e, ...args)
+    }
+    triggerProps.onMouseLeave = (e, ...args) => {
+      this.setTooltipOpen(false, e)
+      _.invoke(triggerElement, 'props.onMouseLeave', e, ...args)
+    }
+
+    return triggerProps
+  }
+
+  getContentProps = (predefinedProps?) => {
+    const contentHandlerProps: Props = {}
+
+    contentHandlerProps.onMouseEnter = (e, contentProps) => {
+      this.setTooltipOpen(true, e)
+      _.invoke(predefinedProps, 'onMouseEnter', e, contentProps)
+    }
+    contentHandlerProps.onMouseLeave = (e, contentProps) => {
+      this.setTooltipOpen(false, e)
+      _.invoke(predefinedProps, 'onMouseLeave', e, contentProps)
+    }
+
+    return contentHandlerProps
+  }
+
+  shouldClose = e => {
```

Comment:
```tsx
shouldClose = e => {
    const containsTarget = _.invoke(e, 'currentTarget.contains', e.relatedTarget) && _.invoke(this.contentRef.current, 'contains', e.relatedTarget)
    
    return !containsTarget
```

Can be simplied with `Lodash`, may be it makes sense to rename it:

```tsx
shouldStayOpen = e => _.invoke(e, 'currentTarget.contains', e.relatedTarget) && _.invoke(this.contentRef.current, 'contains', e.relatedTarget)
```

```

Reply:
Agreed, is more simpler, just `&&` should be replaced with `||`:
```
shouldStayOpen = e => _.invoke(e, 'currentTarget.contains', e.relatedTarget) || _.invoke(this.contentRef.current, 'contains', e.relatedTarget)

```